source/iar/retarget_os_rtos2.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2023 ARM Limited or its affiliates. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cmsis_os2.h"
+#include "cmsis_compiler.h"
+
+//#if defined(_DLIB_THREAD_SUPPORT) && _DLIB_THREAD_SUPPORT > 0
```

Comment:
I'm not sure if this is left in on purpose or is just a leftover, so please check.

Reply:
hm. This was an old PR I had forgot to push. So this is not top of mind for me anymore :)
If _DLIB_THREAD_SUPPORT == 0 you have a libc without multithreading, and the __iar_system/file functions will not be called. Should this cause a compilation failure or not?
I would suspect it is better to get an error, but I don't have an overview of the system to say either way.

Reply:
No, it won't cause compilation failure, but there will be unused functions in the image, since they have the __USED attribute.
A project that uses "OS Interface" component will want to have C library with multithreading hence it would probably make sense to just:
```
#if defined(_DLIB_THREAD_SUPPORT) && _DLIB_THREAD_SUPPORT == 0
  #error "OS Interface requires standard C library with multithreading support enabled."
#endif
```
or something similar. Just an idea...

Reply:
I removed it. This flag wasn't set in modern versions of the libc anyways.